### ISSUE TITLE
feat: deep-linking

### ISSUE THREAD

**Issue by abbey-titcomb** [radicle-dev/radicle-upstream#1512]
[From radicle.community](https://radicle.community/t/http-links-to-open-in-the-client/463)

I have one feature suggestion. It would be cool to have a button to copy the URL that opens the Radicle client app, similarly to how https://t.me/telegram is opened by Telegram, or https://us04web.zoom.us/j/XXX?pwd=XXX in Zoom.

It’d be very convenient to share such links on the internet in general, but my goal is to make a badge in the repo readme with the link to open my project in Radicle. Does it sound like a good idea?

the URL could be something like: https://open.radicle.xyz/<project_id>

Adding to [Feedback Hub](https://github.com/orgs/radicle-dev/projects/19?add_cards_query=is%3Aopen)

**Comment by dragon788** [comment]
Related to #1498 as webapps (PWA/SPA/etc) can also be registered as system protocol handlers in some cases.

**Comment by abbey-titcomb** [comment]
@cloudhead said we could do this with xdg-open for Linux, @rudolfs - any idea what we could do for Mac?

**Comment by rudolfs** [comment]
I think the sanest cross-platform way to get Upstream to open if a user clicks a special link somewhere in a browser would be registering a custom protocol. Basically what @dragon788 mentioned above and @xla [suggested back in the day](https://github.com/radicle-dev/radicle-upstream/issues/840#issuecomment-703491425).

**Comment by rudolfs** [comment]
> the URL could be something like: https://open.radicle.xyz/<project_id>

I'm not entirely sure what's the benefit of this additional layer of wrapping. Could you elaborate on that?

**Comment by cloudhead** [comment]
The additional layer is definitely not needed for now.

**Comment by abbey-titcomb** [comment]
Saw another tweet that references this today - https://twitter.com/grin_io/status/1352630590173016066?s=20

Would be cool to try to build this in soon !

**Comment by NunoAlexandre** [comment]
@abbey-titcomb I'd be interested in picking this up. For that, I would like us to scope this and come up with a v1 concept. My idea is for this fist iteration is to support deep linking for projects only.

Now, if users contain the linked project locally, we would simply open the screen with that project's overview.
If the user does \**not\** contain a local copy of that project, they'd first need to follow it. For that, I'd say it would be nice to show a modal informing the user that the project they requested to open is not found locally and asking the user whether they want to follow it. Alternatively, we could just fill in the value in the search box, but that would provide no context for unfamiliar users.

Let me have your thoughts :seedling:

**Comment by NunoAlexandre** [comment]
So I gave this a quick spin. Conceptually, this is not a big deal: we will have the app registering, say, `rad://` as a custom protocol that it will become responsible to handle, meaning that the OS should essentially pass on requests to that protocol to the app. In the app, we will intercept those requests, parse them, and handle them accordingly.

Now, I tried it out to have the app owning requests to that custom protocol but it doesn't seem to work out of the box on Linux and/or with `AppImage`. More specifically, I am facing [this exact same reported problem](https://github.com/electron-userland/electron-builder/issues/4035).

I didn't have a chance to try this on a macOS machine yet.

One observation arises to me though. This deep-linking feature, from a user coming from the web, would make its way if the user would already have `Upstream` installed on their machine.

_bookmarks_: 
- https://discuss.atom.io/t/enable-deep-linking/27518
- https://github.com/electron-userland/electron-builder/issues/4035
